core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/definitions/ldap/implementations/UnboundidLdapConnectionPoolProvider.scala

```diff
@@ -186,6 +176,31 @@ object UnboundidLdapConnectionPoolProvider extends Logging {
       .asyncBoundary
   }
 
+  private def testBinding(connectionConfig: LdapConnectionConfig) = {
+    val serverSet = createLdapServerSet(connectionConfig)
+    val bindReq = bindRequest(connectionConfig.bindRequestUser)
+    val maxRetries = if (connectionConfig.ignoreLdapConnectivityProblems) 0 else 5
+    val connectionTimeout: FiniteDuration = if (connectionConfig.ignoreLdapConnectivityProblems) {
+      List(2 seconds, connectionConfig.connectionTimeout.value).min
+    } else {
+      connectionConfig.connectionTimeout.value
+    }
+    val resource =
+      Resource.make(
+        Task(serverSet.getConnection).timeout(connectionTimeout)
+      ) { conn =>
+        Task(conn.close())
+      }
+    retryBackoff(
+      source = resource.use { connection =>
+        Task(connection.bind(bindReq)).timeout(connectionConfig.requestTimeout.value)
```

Comment:
in `ldapOptions` we set the connect timeout. Here we set some timeout at the Task level. Why?

Reply:
Oh, I recall - the invalid hostname. But shouldn't we use our `HostnameResolver` to handle it? 

I see that UnboundId resolves the host really late. See `LDAPCOnnection` class:
```
final InetAddress inetAddress;
    try
    {
      inetAddress = connectionOptions.getNameResolver().getByName(host);
    }    
```

Maybe we can do it by ourselves before passing the hostname to the UnboundID's serverSet?